BankApplication/src/com/luxoft/bankapp/model/CheckingAccount.java

```diff
@@ -0,0 +1,121 @@
+package com.luxoft.bankapp.model;
+
+
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.luxoft.bankapp.handling_exceptions.OverDraftLimitExceededException;
+
+public class CheckingAccount extends AbstractAccount {
+	
+	private final static Logger LOG = LoggerFactory.getLogger(CheckingAccount.class);
+
+	private float overdraft;
+	private String accountType="CheckingAccount";
+
+
+	public CheckingAccount(float balance) {
+		super(balance);
+		this.overdraft=0;
+	}
+	
+	
+	public float getOverdraft() {
+		return overdraft;
+	}
+
+
+	public CheckingAccount(float balance, Client client) {
+		super(balance);
+		overdraft=client.getInitialOverdraft();
+		if(overdraft<0)
+		{
+			LOG.warn("Illegal argument");
+			throw new IllegalArgumentException(Float.toString(overdraft));
+		}
+		if(balance<-overdraft)
+		{
+			LOG.warn("Illegal argument");
+			throw new IllegalArgumentException(Float.toString(balance));
+		}	
+
+	}
+	
+	
+	public void setOverdraft(float overdraft) throws IllegalArgumentException
+	{
+		if(overdraft<0)
+		{
+			LOG.warn("Illegal argument");
```

Comment:
Here would be good to provide some more info: what is the argument value and why is it illigal.